crates/matrix-sdk-common/src/deserialized_responses.rs

```diff
@@ -684,9 +701,7 @@ pub enum UnableToDecryptReason {
 
     /// Decryption failed because we're missing the megolm session that was used
     /// to encrypt the event.
-    ///
-    /// TODO: support withheld codes?
-    MissingMegolmSession,
+    MissingMegolmSession(Option<WithheldReason>),
```

Comment:
I think this change is going to break deserialization of existing  TimelineEvents? Needs some serde magic. Or a test that proves it's ok

Reply:
I though we said it was fine for now, looking at that

Reply:
Here https://github.com/matrix-org/matrix-rust-sdk/pull/4305/commits/15297fb3b7646681fa3b3c6033a5ef5a6ad390be